xarray/testing.py

```diff
@@ -140,3 +147,57 @@ def assert_allclose(a, b, rtol=1e-05, atol=1e-08, decode_bytes=True):
     else:
         raise TypeError('{} not supported by assertion comparison'
                         .format(type(a)))
+
+
+def assert_allclose_with_nan(a, b, rtol=1e-05, atol=1e-08, decode_bytes=True):
+    """Like assert_allclose, but except for nan.
```

Comment:
Can we consolidate this into `assert_allclose()`, like `numpy.testing.assert_allclose()`, which has an `equal_nan=True` argument?

Reply:
This will overwrite by #1883. Sorry for letting you review outdated code.

Reply:
Can we still move this into `assert_allclose()`? Most of the time, I think we do want to use `equal_nan=True`.

Reply:
I noticed the current `assert_allclose` assumes `equal_nan=True` by default. I dropped `assert_allclose_with_nan`.